pip/qsharp/_qsharp.py

```diff
@@ -169,6 +179,13 @@ def eval(source: str) -> Any:
     ipython_helper()
 
     def callback(output: Output) -> None:
+        if _in_jupyter:
+            try:
+                display(output)
+                return
+            except:
+                # If IPython is not available, fall back to printing the output
+                pass
```

Comment:
The old behavior of `qsharp.eval` was intentional. May be debatable whether it's the "ideal" behavior but definitely intentional as seen in the [sample notebook](https://github.com/microsoft/qsharp/blob/main/samples/notebooks/sample.ipynb):

![image](https://github.com/user-attachments/assets/90dac782-6fb6-4fe6-8aee-49132049c1b8)
 


If we feel like we _have_ to change this behavior, I think we should take more care to refactor this code so that it's not duplicated with the code in `_ipython.py` (which I see, also got a new bugfix we'd want to include here): https://github.com/microsoft/qsharp/blob/78ad29efd6b821cf8d1f98e44bbdd50a313dd91f/pip/qsharp/_ipython.py#L19-L29

Reply:
> The old behavior of qsharp.eval was intentional.

Why was it intentional that they appear differently. What's the rationale for that? I had assumed that was a mistake/oversight.

Reply:
Ping @minestarks . Still not clear on why it was intentional, and if there are benefits to that behavior. (i.e. if should revert changes around it)

Reply:
I don't think there was a compelling reason for the old behavior, I just wanted it to behave more "console-y" when `qsharp.eval` was used as opposed to the ipython magic `%%qsharp`.

I'm ok changing it but please note my comment on code duplication. The same `display` logic is implemented in `_ipython.py` and I think we should leverage it.

Reply:
The code in _ipython.py is just for `@register_cell_magic`, in which case if you're in a `%%qsharp` cell we __known___ you're in a notebook context, right? So I don't think that needs the display logic from _qsharp.py, which is specifically to detect if in a text based UI (e.g. IPython in the console) versus a GUI.

Or am I missing something?

Reply:
Will address later any changes needed on the above question.